controllers/memberoperatorconfig/memberoperatorconfig_controller.go

```diff
@@ -48,5 +51,58 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	reqLogger := log.FromContext(ctx)
 	reqLogger.Info("Reconciling MemberOperatorConfig")
 
-	return reconcile.Result{}, loadLatest(r.Client)
+	crtConfig, err := loadLatest(r.Client)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.handleAutoscalerDeploy(reqLogger, crtConfig, request.Namespace); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.handleUserPodsWebhookDeploy(reqLogger, crtConfig, request.Namespace); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *Reconciler) handleAutoscalerDeploy(logger logr.Logger, cfg Configuration, namespace string) error {
+	if cfg.Autoscaler().Deploy() {
+		logger.Info("(Re)Deploying autoscaling buffer")
+		if err := autoscaler.Deploy(r.Client, r.Client.Scheme(), namespace, cfg.Autoscaler().BufferMemory(), cfg.Autoscaler().BufferReplicas()); err != nil {
+			logger.Error(err, "failed to deploy autoscaling buffer")
```

Comment:
I think you should just wrap the error with `"failed to deploy autoscaling buffer"` and return it, and let the caller log the whole error

Reply:
I initially did that but the autoscaler errors are already wrapped so I felt it was kind of unnecessary to wrap it again: https://github.com/codeready-toolchain/member-operator/blob/master/pkg/autoscaler/autoscaler.go#L30

Reply:
but in that case, we don't need to log the error here, since it should already be logged by the caller, shouldn't it?

Reply:
Looking at https://github.com/codeready-toolchain/member-operator/blob/master/pkg/autoscaler/autoscaler.go#L30 it doesn't look like it's logged. The error is just wrapped and returned.

Reply:
well, I assume it's logged by the reconciler? or am I missing something?

Reply:
but we can keep it like that for now and see how it goes when an error occurs.

Reply:
Removed the logger.Error calls as discussed: 728a4a3987b91c5ff92c57392dd1c8147a025335